java/ql/src/experimental/Security/CWE/CWE-691/InsecureClassLoading.java

```diff
@@ -0,0 +1,30 @@
+
+import android.app.Application;
+import android.content.Context;
+import android.content.pm.PackageInfo;
+import android.os.Bundle;
+
+public class InsecureClassLoading extends Application {
+  @Override
+  public void onCreate() {
+    super.onCreate();
+    invokePlugins();
+  }
+
+  private void invokePlugins() {
+    for (PackageInfo info : getPackageManager().getInstalledPackages(0)) {
+      String packageName = info.packageName;
+      Bundle meta = info.applicationInfo.metaData;
+      if (packageName.startsWith("somename.plugin.")) {
+        try {
+          Context packageContext =
+              createPackageContext(packageName, CONTEXT_INCLUDE_CODE | CONTEXT_IGNORE_SECURITY);
```

Comment:
Is the IGNORE_SECURITY flag here fundamental or merely incidental to the root problem?